src/services/passport.ts

```diff
@@ -0,0 +1,84 @@
+import axios, { AxiosInstance } from "axios";
+import { PASSPORT_API_KEY } from "../../config.json";
+import {
+  LinkAccountBody,
+  LinkedAccountsResponse,
+  LinkedAddressesResponse,
+} from "../types/passport";
+
+const baseURL = "https://galactic-passport-server.herokuapp.com/v1";
+
+class PassportAPI {
+  passportClient: AxiosInstance;
+
+  constructor() {
+    this.passportClient = axios.create({
+      baseURL,
+      headers: {
+        "X-GP-API-KEY": PASSPORT_API_KEY,
+        "Content-Type": "application/json",
+      },
+    });
+  }
+
+  // get requests
+
+  // get wallets of discord id
+  getWalletsByDiscordId = async (discordId: string): Promise<string[]> => {
+    const res = (
+      await this.passportClient.get(
+        `/linked_addresses?platform_ids=discord_${discordId}`
+      )
+    ).data as LinkedAddressesResponse;
+
+    const wallets = res[discordId].wallets;
+
+    if (wallets == undefined) {
+      throw new Error("Wallets returned are undefined");
+    }
+
+    return wallets.map((walletObj) => walletObj.address);
+  };
+
+  // get discord id of wallet
+  getDiscordIdByWallet = async (address: string): Promise<string> => {
+    const res = (
+      await this.passportClient.get(
+        `/linked_accounts?platforms=discord&addresses=${address}`
+      )
+    ).data as LinkedAccountsResponse;
+
+    if (res.accounts == undefined) {
+      throw new Error("Accounts returned are undefined");
+    }
+
+    const discordId = res.accounts[0].accountId;
```

Comment:
Are there going to be any problems if the user links one address to multiple discord accounts? 

Like if they try to delete, it could grab the wrong account to delete, right?

Reply:
Does the bot not have access to the user's `discordId`?